gitlab/resource_gitlab_project_approval_rule.go

```diff
@@ -0,0 +1,254 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+// https://docs.gitlab.com/ee/api/merge_request_approvals.html#create-project-level-rule
+
+func resourceGitlabProjectApprovalRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabProjectApprovalRuleCreate,
+		Read:   resourceGitlabProjectApprovalRuleRead,
+		Update: resourceGitlabProjectApprovalRuleUpdate,
+		Delete: resourceGitlabProjectApprovalRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"approvals_required": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"user_ids": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+				Set:      schema.HashInt,
+			},
+			"group_ids": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+				Set:      schema.HashInt,
```

Comment:
I unfortunately can't find the reference for this, but I saw someone from HashiCorp mention that it is better to program providers with attributes of type `list` instead of `set`, because it makes the Terraform diff more readable for users.

Reply:
I've updated these to use `schema.TypeList`